In [ ]:
import numpy as np
import pandas as pd

In [ ]:
from google.colab import files
uploaded = files.upload()


Saving WA_Fn-UseC_-Telco-Customer-Churn.csv to WA_Fn-UseC_-Telco-Customer-Churn.csv


In [ ]:
df=pd.read_csv('WA_Fn-UseC_-Telco-Customer-Churn.csv')

In [ ]:
df.head()

customerID  gender  SeniorCitizen  ... MonthlyCharges TotalCharges  Churn
0  7590-VHVEG  Female              0  ...          29.85        29.85     No
1  5575-GNVDE    Male              0  ...          56.95       1889.5     No
2  3668-QPYBK    Male              0  ...          53.85       108.15    Yes
3  7795-CFOCW    Male              0  ...          42.30      1840.75     No
4  9237-HQITU  Female              0  ...          70.70       151.65    Yes

[5 rows x 21 columns]

#Nomor 2 poin 1


---



In [ ]:
df['MonthlyCharges'].describe()

count    7043.000000
mean       64.761692
std        30.090047
min        18.250000
25%        35.500000
50%        70.350000
75%        89.850000
max       118.750000
Name: MonthlyCharges, dtype: float64

In [ ]:
df['MonthlyCharges'].mode()

0    20.05
dtype: float64

Pada kolom diatas, nilai rata-rata dari keseluruhan yang ada adalah 64.76, dengan nilai terkecil 18.25 dan nilai terbesar 118.75. Nilai median yang dimiliki kolom tersebut adalah 70.35 dengan modus 20.05. Lalu standar deviasinya adalah 30.09

In [ ]:
df['tenure'].describe()

count    7043.000000
mean       32.371149
std        24.559481
min         0.000000
25%         9.000000
50%        29.000000
75%        55.000000
max        72.000000
Name: tenure, dtype: float64

In [ ]:
df['tenure'].mode()

0    1
dtype: int64

Pada kolom diatas, nilai rata-rata dari keseluruhan yang ada adalah 32.37, dengan nilai terkecil 0 dan nilai terbesar 72. Nilai median yang dimiliki kolom tersebut adalah 29 dengan modus 1. Lalu standar deviasinya adalah 24.56

In [ ]:
newTC = df['TotalCharges'].apply(pd.to_numeric, errors='coerce')



In [ ]:
newTC.describe()

count    7032.000000
mean     2283.300441
std      2266.771362
min        18.800000
25%       401.450000
50%      1397.475000
75%      3794.737500
max      8684.800000
Name: TotalCharges, dtype: float64

In [ ]:
df['TotalCharges'].mode()

0        
1    20.2
dtype: object

Pada kolom diatas, nilai rata-rata dari keseluruhan yang ada adalah 2283.30, dengan nilai terkecil 18.8 dan nilai terbesar 8684.80. Nilai median yang dimiliki kolom tersebut adalah 1397.48 dengan modus 20.2. Lalu standar deviasinya adalah 2266.77

In [ ]:
df.groupby(['gender','Churn'])['Churn'].count()

gender  Churn
Female  No       2549
        Yes       939
Male    No       2625
        Yes       930
Name: Churn, dtype: int64

Churn rate untuk yang Female adalah 26.9% dan Male adalah 26.1%

In [ ]:
df.groupby(['Partner','Churn'])['Churn'].count()

Partner  Churn
No       No       2441
         Yes      1200
Yes      No       2733
         Yes       669
Name: Churn, dtype: int64

Churn rate untuk yang bukan partner adalah 33% dan yang partner adalah 19.6%

In [ ]:
df.groupby(['Dependents','Churn'])['Churn'].count()

Dependents  Churn
No          No       3390
            Yes      1543
Yes         No       1784
            Yes       326
Name: Churn, dtype: int64

In [ ]:
Churn rate untuk yang tidak dependen adalah 31.3% dan yang dependen adalah 15.5%

In [ ]:
df.groupby(['InternetService','Churn'])['Churn'].count()

InternetService  Churn
DSL              No       1962
                 Yes       459
Fiber optic      No       1799
                 Yes      1297
No               No       1413
                 Yes       113
Name: Churn, dtype: int64

Churn rate untuk yang menggunakan DSL adalah 19%, Fiber Optic 41.9%, dan yang tidak menggunakan internet 7.4%


In [ ]:
df.groupby(['Contract','Churn'])['Churn'].count()

Contract        Churn
Month-to-month  No       2220
                Yes      1655
One year        No       1307
                Yes       166
Two year        No       1647
                Yes        48
Name: Churn, dtype: int64

Churn rate yang memiliki kontrak bulan ke bulan adalah 42.7%, setahun 11.3%, dan dua tahun 2.8% 

In [ ]:
df.groupby(['PaymentMethod','Churn'])['Churn'].count()

PaymentMethod              Churn
Bank transfer (automatic)  No       1286
                           Yes       258
Credit card (automatic)    No       1290
                           Yes       232
Electronic check           No       1294
                           Yes      1071
Mailed check               No       1304
                           Yes       308
Name: Churn, dtype: int64

Churn rate yang  menggunakan bank transfer adalah 16.7%, credit card 15,2% , electronic check 45%, dan mailed check 19.1%